### PR TITLE
Make percentage_format_string support str.format() syntax

### PR DESCRIPTION
**Pull request by iamxeph**
## Description

This PR make percentage_format_string support str.format() syntax in addition to old %-format syntax.

## Checklist

- [X] I have reviewed the [contributing guidelines](https://github.com/jertel/elastalert2/blob/master/CONTRIBUTING.md).
- [X] I have included unit tests for my changes or additions.
- [X] I have successfully run `make test-docker` with my changes.
- [X] I have updated the [documentation](https://elastalert2.readthedocs.io).
- [X] I have updated the [changelog](https://github.com/jertel/elastalert2/blob/master/CHANGELOG.md).

### CODE REVIEW

File: elastalert/ruletypes.py [hunk 1]
```diff
@@ -1308,3 +1308,9 @@ def percentage_violation(self, match_percentage):
         if 'min_percentage' in self.rules and match_percentage < self.rules['min_percentage']:
             return True
         return False
+
+    def format_string(self, format_config, target_value):
```
Comment (jertel) [inline elastalert/ruletypes.py:1312]: What do you think about moving this format_string() into a util class so that it doesn't get repeated here?

Comment (iamxeph) [reply elastalert/ruletypes.py:1312]: Working on it

Comment (iamxeph) [reply elastalert/ruletypes.py]: Moved format_string() to util.py and added tests for the function accordingly

**Review by jertel — APPROVED** [review]
Thanks!

**Pull request merged**